addons/mail/static/tests/discuss/pinned_messages_tests.js

```diff
@@ -0,0 +1,125 @@
+/** @odoo-module */
+
+import { nextTick, patchWithCleanup } from "@web/../tests/helpers/utils";
+import { click, start, startServer } from "@mail/../tests/helpers/test_utils";
+
+QUnit.module("pinned messages");
+
+QUnit.test("Pin message", async (assert) => {
+    const pyEnv = await startServer();
+    const channelId = pyEnv["mail.channel"].create({ name: "General" });
+    pyEnv["mail.message"].create({
+        body: "Hello world!",
+        model: "mail.channel",
+        res_id: channelId,
+    });
+    const { openDiscuss } = await start();
+    await openDiscuss(channelId);
+    await click(".o-mail-Discuss-header button[title='Pinned messages']");
+    assert.containsOnce(
+        $,
+        ".o-mail-PinnedMessagesPanel:contains(This channel doesn't have any pinned messages.)"
```

Comment:
If you change the string to "conversation" regardless of thread, don't forget to adapt this test ;)